src/pages/EntityDetails/Model/Secrets/SecretContent/SecretContent.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  Button,
+  CodeSnippet,
+  Icon,
+  Modal,
+  Tooltip,
+  ActionButton,
+  Notification,
+} from "@canonical/react-components";
+import { Form, Formik } from "formik";
+import { useParams } from "react-router-dom";
+import usePortal from "react-useportal";
+
+import type { EntityDetailsRoute } from "components/Routes/Routes";
+import RevisionField from "components/secrets/RevisionField";
+import { useGetSecretContent } from "juju/apiHooks";
+import { actions as jujuActions } from "store/juju";
+import {
+  getSecretByURI,
+  getSecretsContent,
+  getSecretsContentErrors,
+  getSecretsContentLoaded,
+  getSecretsContentLoading,
+  getModelUUIDFromList,
+  getModelByUUID,
+} from "store/juju/selectors";
+import { useAppSelector, useAppDispatch } from "store/store";
+
+export enum Label {
+  MODAL_TITLE = "Secret values",
+  SHOW = "Show content",
+  SUBMIT = "View",
+}
+
+type Props = {
+  label?: string;
+  secretURI: string;
+};
+
+const SecretContent = ({ label, secretURI }: Props) => {
+  const { userName, modelName } = useParams<EntityDetailsRoute>();
+  const dispatch = useAppDispatch();
+  const modelUUID = useAppSelector(getModelUUIDFromList(modelName, userName));
+  const wsControllerURL = useAppSelector((state) =>
+    getModelByUUID(state, modelUUID),
+  )?.wsControllerURL;
+  const secret = useAppSelector((state) =>
+    getSecretByURI(state, modelUUID, secretURI),
+  );
+  const content = useAppSelector((state) =>
+    getSecretsContent(state, modelUUID),
+  );
+  const contentError = useAppSelector((state) =>
+    getSecretsContentErrors(state, modelUUID),
+  );
+  const contentLoaded = useAppSelector((state) =>
+    getSecretsContentLoaded(state, modelUUID),
+  );
+  const contentLoading = useAppSelector((state) =>
+    getSecretsContentLoading(state, modelUUID),
+  );
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const getSecretContent = useGetSecretContent(userName, modelName);
+
+  return (
+    <>
+      <Tooltip message="View secret values">
+        <Button
+          appearance="base"
+          className="is-small"
+          onClick={openPortal}
+          type="button"
+          hasIcon
+        >
+          <Icon name="show">{Label.SHOW}</Icon>
+          {label ? <span>{label}</span> : null}
```

Comment:
Was wondering if it might look more user-friendly if only the icon is clickable instead of the icon + label?

Reply:
Done.